### PR TITLE
Add extra search paths for the ISPC binary

### PR DESCRIPTION
**Pull request by urkle**
- have it look in the checkout if the user manually downloads the binary to those locations

### CODE REVIEW

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -25,6 +25,13 @@ else()
 				set(ispcVersion ${curVersion})
 			endif()
 		endforeach()
+		list(APPEND ispcInstallPath ${ISPC_TEXCOMP_DIR}/ISPC/win)
+	endif()
+	if(CMAKE_SYSTEM MATCHES "Linux")
```
Comment (akb825) [inline lib/CMakeLists.txt:30]: Would be cleaner to use elseif() rather than else() followed by if().

File: lib/CMakeLists.txt [hunk 2]
```diff
@@ -25,6 +25,13 @@ else()
...
+		list(APPEND ispcInstallPath ${ISPC_TEXCOMP_DIR}/ISPC/linux)
```
Comment (akb825) [inline lib/CMakeLists.txt:31]: ispcInstallPath isn't set first, so would be better to use set() rather than list(APPEND).

**Comment by akb825** [comment]
I'm a little wary of this change:
* The submodules are largely implementation details, and aren't really meant to be modified apart from updating as needed.
* For the general user, they'd have to explicitly look in the README of the submodule and follow the instructions to manually download ISPC to those specific locations.
* Any changes to the directory structure of ISPCTextureCompressor could also cause this to break in a way that CI won't catch, though admittedly this isn't super likely.
* ISPC isn't used only for ISPCTextureCompressor but also for bc7enc_rdo, so it seems a little arbitrary if the binary is "owned" by one specific submodule.

The intended way for users to specify a non-default install location of ISPC is to set `CUTTLEFISH_ISPC_PATH` to the path of the executable. This doesn't require modifying the directory structure in the repository and avoids the various issues noted above. Is this problematic for your use case? If the current solution isn't sufficient for your usage there is likely another approach we can take.

File: lib/CMakeLists.txt (hunk 2 above)
Comment (urkle) [reply lib/CMakeLists.txt:31]: it doesn't matter too much actually as the cmake docs state if the var is undefined it treats it as an empty string.

File: lib/CMakeLists.txt (hunk 1 above)
Comment (urkle) [reply lib/CMakeLists.txt:30]: yes it would :)

**Comment by urkle** [comment]
> The intended way for users to specify a non-default install location of ISPC is to set CUTTLEFISH_ISPC_PATH to the path of the executable

🤦 it was right there in front of my face and didn't realize it :)   Honestly that'll work fine.. I'll close this PR.

File: lib/CMakeLists.txt (hunk 2 above)
Comment (akb825) [reply lib/CMakeLists.txt:31]: While true, it opens you up to the possibility that an earlier CMakeLists.txt defined the variable.

**Pull request closed without merging**